deltacat/compute/compactor_v2/private/compaction_utils.py

```diff
@@ -173,6 +171,33 @@ def _build_uniform_deltas(
     )
 
 
+def _group_uniform_deltas(params: CompactPartitionParams, uniform_deltas):
+    num_deltas = len(uniform_deltas)
+    num_rounds = params.num_rounds
+    if num_rounds == 1:
+        return [uniform_deltas]
+    assert (
+        num_rounds <= num_deltas
+    ), f"{params.num_rounds} rounds should be less than the number of uniform deltas, which is {len(uniform_deltas)}"
+    assert (
+        params.hash_bucket_count != 1
+    ), "Hash bucket count is set to 1, we should not be grouping uniform deltas into batches"
+    if num_rounds >= num_deltas:
+        return [uniform_deltas]
+    sublist_length = num_deltas // num_rounds
```

Comment:
Another assertion of num_rounds not equal to 0 nor negative number and will always be an integer to make sure when we override in scaling params in the future, we didn't put a unreasonable number for `num_rounds` accidentally and break the whole compaction job.